lib/helper/storage.js

```diff
@@ -0,0 +1,172 @@
+'use babel';
+
+import { basename, dirname, leadingslashit, trailingslashit, unleadingslashit, untrailingslashit, normalize, cleanJsonString } from './../helper/format.js';
+import { decrypt, encrypt } from './../helper/secure.js';
+import { throwErrorIssue44 } from './../helper/issue.js';
+
+const atom = global.atom;
+const server_config = require('./../config/server-schema.json');
+const folder_config = require('./../config/folder-schema.json');
+
+class Storage {
+
+    constructor() {
+      const self = this;
+
+      self.servers = [];
+      self.folders = [];
+      self.version = '';
+      self.password = null;
+      self.loaded = false;
+    }
+
+    setPassword(password) {
+        const self = this;
+
+        self.password = password;
+    }
+
+    getPassword() {
+        const self = this;
+
+        return self.password;
+    }
+
+    hasPassword() {
+        const self = this;
+
+        return self.password !== null;
+    }
+
+    load(reload = false) {
+        const self = this;
+
+        if(!self.loaded || reload) {
+            let configText = atom.config.get('ftp-remote-edit.config');
+            if (configText) {
+              try {
+                configText = JSON.stringify(JSON.parse(cleanJsonString(configText)), null, 4);
+              } catch (e) {
+                configText = decrypt(self.password, configText);
+              }
+
+              try {
+                  configArray = JSON.parse(cleanJsonString(configText));
+
+                  if(configArray.version instanceof String) {
+                      self.version = configArray.version;
+                  } else {
+                      self.version = atom.packages.getActivePackage('ftp-remote-edit').metadata.version;
+                  }
+
+                  if(typeof configArray.servers === 'undefined' && configArray instanceof Array) {
```

Comment:
I would outsource the check to an extra migrate function, because this can be used for other migrations in the future.